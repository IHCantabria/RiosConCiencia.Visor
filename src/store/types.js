//DATAHUB

//Mutations
export const M_SET_PRODUCTS = "datahub/M_SET_PRODUCTS";
export const M_SET_SOURCES = "datahub/M_SET_SOURCES";
export const M_SET_PRODUCTMOMENTS = "datahub/M_SET_PRODUCTMOMENTS";
export const M_SET_PRODUCTVARIABLETYPES = "datahub/M_SET_PRODUCTVARIABLETYPES";
export const M_UPDATE_PRODUCTS_SELECTED = "datahub/M_UPDATE_PRODUCTS_SELECTED";

//Actions
export const A_FETCH_DATAHUB_PRODUCTS = "datahub/A_FETCH_DATAHUB_PRODUCTS";
export const A_FETCH_DATAHUB_SOURCES = "datahub/A_FETCH_DATAHUB_SOURCES";
export const A_FETCH_DATAHUB_PRODUCTMOMENTS =
  "datahub/M_SET_DATAHUB_PRODUCTMOMENTS";
export const A_FETCH_DATAHUB_PRODUCTVARIABLETYPES =
  "datahub/A_FETCH_DATAHUB_PRODUCTVARIABLETYPES";
export const A_SET_CLEAR_PRODUCTS = "datahub/A_SET_CLEAR_PRODUCTS";
export const A_UPDATE_PRODUCTS_SELECTED = "datahub/A_UPDATE_PRODUCTS_SELECTED";

//Getters
export const G_GET_PRODUCTS_SORTED = "datahub/G_GET_PRODUCTS_SORTED";
export const G_GET_PRODUCTS_SELECTED = "datahub/G_GET_PRODUCTS_SELECTED";
export const G_GET_PRODUCTS_COUNT = "datahub/G_GET_PRODUCTS_COUNT";
export const G_GET_PRODUCT_BY_MAPID = "datahub/G_GET_PRODUCT_BY_MAPID";
export const G_GET_SOURCE_NAMESHORT = "datahub/G_GET_SOURCE_NAMESHORT";
export const G_GET_SOURCE_TYPE = "datahub/G_GET_SOURCE_TYPE";
export const G_GET_PRODUCTMOMENTS_SORTED =
  "datahub/G_GET_PRODUCTMOMENTS_SORTED";
export const G_GET_PRODUCTVARIABLETYPES_BY_PRODUCTMOMENTS =
  "datahub/G_GET_PRODUCTVARIABLETYPES_BY_PRODUCTMOMENTS";

//APISENSORS

//Mutations
export const M_SET_SENSORS = "apiSensors/M_SET_SENSORS";
export const M_SET_BUOYS = "apiSensors/M_SET_BUOYS";
export const M_SET_STATIONS = "apiSensors/M_SET_STATIONS";
export const M_SET_BUOYTYPES = "apiSensors/M_SET_BUOYTYPES";
export const M_SET_STATIONTYPES = "apiSensors/M_SET_STATIONTYPES";

//Actions
export const A_FETCH_APISENSORS_BUOYS = "apiSensors/A_FETCH_APISENSORS_BUOYS";
export const A_FETCH_APISENSORS_STATIONS =
  "apiSensors/A_FETCH_APISENSORS_STATIONS";
export const A_FETCH_APISENSORS_BUOYTYPES =
  "apiSensors/A_FETCH_APISENSORS_BUOYTYPES";
export const A_FETCH_APISENSORS_STATIONTYPES =
  "apiSensors/A_FETCH_APISENSORS_STATIONTYPES";
export const A_SET_SENSORS = "apiSensors/A_SET_SENSORS";
export const A_SET_CLEAR_SENSORS = "apiSensors/A_SET_CLEAR_SENSORS";

//Getters
export const G_GET_SENSORS_SORTED = "apiSensors/G_GET_SENSORS_SORTED";
export const G_GET_SENSORS_SELECTED = "apiSensors/G_GET_SENSORS_SELECTED";
export const G_GET_SENSORS_COUNT = "apiSensors/G_GET_SENSORS_COUNT";
export const G_GET_BUOYTYPE_NAME = "apiSensors/G_GET_BUOYTYPE_NAME";

//COORDS

//Mutations
export const M_SET_SEARCH_COORDS = "coords/M_SET_SEARCH_COORDS";
export const M_SET_GO_TO_COORDS = "coords/M_SET_GO_TO_COORDS";

//Actions
export const A_SET_SEARCH_COORDS = "coords/A_SET_SEARCH_COORDS";
export const A_SET_GO_TO_COORDS = "coords/A_SET_GO_TO_COORDS";

//FILTERS

//Mutations
export const M_SET_FILTER_MOMENT = "filters/M_SET_FILTER_MOMENT";
export const M_SET_FILTER_VARIABLE = "filters/M_SET_FILTER_VARIABLE";
export const M_SET_FILTER_BUOYSTYPE = "filters/M_SET_FILTER_BUOYSTYPE";
export const M_SET_FILTER_STATIONSTYPE = "filters/M_SET_FILTER_STATIONSTYPE";
export const M_SET_FILTER_ONLY_SPATIAL_FILTERS =
  "filters/M_SET_FILTER_ONLY_SPATIAL_FILTERS";
export const M_SET_FILTER_ALLOW_PRODUCTS =
  "filters/M_SET_FILTER_ALLOW_PRODUCTS";
export const M_SET_FILTER_ALLOW_BUOYS = "filters/M_SET_FILTER_ALLOW_BUOYS";
export const M_SET_FILTER_ALLOW_STATIONS =
  "filters/M_SET_FILTER_ALLOW_STATIONS";

//Actions
export const A_SET_FILTER_MOMENT = "filters/A_SET_FILTER_MOMENT";
export const A_SET_FILTER_VARIABLE = "filters/A_SET_FILTER_VARIABLE";
export const A_SET_FILTER_BUOYSTYPE = "filters/A_SET_FILTER_BUOYSTYPE";
export const A_SET_FILTER_STATIONSTYPE = "filters/A_SET_FILTER_STATIONSTYPE";
export const A_SET_FILTER_ONLY_SPATIAL_FILTERS =
  "filters/A_SET_FILTER_ONLY_SPATIAL_FILTERS";
export const A_SET_FILTER_ALLOW_PRODUCTS = "filters/_SET_FILTER_ALLOW_PRODUCTS";
export const A_SET_FILTER_ALLOW_BUOYS = "filters/A_SET_FILTER_ALLOW_BUOYS";
export const A_SET_FILTER_ALLOW_STATIONS =
  "filters/A_SET_FILTER_ALLOW_STATIONS";

//GETTERS
export const G_GET_FILTERS_READY = "filters/G_GET_FILTERS_READY";
