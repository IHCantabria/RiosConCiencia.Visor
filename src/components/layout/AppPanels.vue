<template>
  <div class="app-info">
    <div class="panel-controls">
      <div
        class="show-button show-button__left"
        title="Sensors"
        data-cy="sensorsToggle"
        @click="sensorsOpen = !sensorsOpen"
      >
        <object
          :data="require('@/assets/sensors.svg')"
          type="image/svg+xml"
          style="height:100%; pointer-events: none;"
        />
      </div>
      <div
        class="show-button show-button__right"
        title="Products"
        data-cy="productsToggle"
        @click="productsOpen = !productsOpen"
      >
        <object
          :data="require('@/assets/products.svg')"
          type="image/svg+xml"
          style="height:100%; pointer-events: none;"
        />
      </div>
    </div>
    <div class="transition-sensors">
      <transition name="slide" class="sensors">
        <div
          v-show="isSensorsOpen"
          data-cy="sensorsPanel"
          :class="[
            'side-panel',
            'side-panel__sensors',
            isSensorsExpanded ? 'side-panel__expanded' : 'side-panel__reduced'
          ]"
        >
          <div class="header-panel">
            <div
              :class="[
                'header-buttons',
                'header-buttons__left',
                isSensorsExpanded ? 'header-buttons__expanded' : ''
              ]"
            >
              <i
                data-cy="sensorsTogglePanel"
                class="fas fa-times header-button"
                @click="sensorsOpen = !sensorsOpen"
              />
              <i
                :class="[
                  'header-button',
                  isSensorsExpanded ? 'fas fa-angle-down' : 'fas fa-angle-up'
                ]"
                data-cy="sensorsTogglePanelExpanded"
                @click="sensorsExpanded = !sensorsExpanded"
              />
            </div>
          </div>
          <panel-sensors
            :table-max-height="computedTableMaxHeightSensors"
            :table-elements="computedTableMaxItemsSensors"
          ></panel-sensors>
        </div>
      </transition>
    </div>
    <div class="transition-products">
      <transition name="slide" class="products">
        <div
          v-show="isProductsOpen"
          data-cy="productsPanel"
          :class="[
            'side-panel',
            'side-panel__products',
            isProductsExpanded ? 'side-panel__expanded' : 'side-panel__reduced'
          ]"
        >
          <div class="header-panel">
            <div
              :class="[
                'header-buttons',
                'header-buttons__right',
                isProductsExpanded ? 'header-buttons__expanded' : ''
              ]"
            >
              <i
                data-cy="productsTogglePanel"
                class="fas fa-times header-button"
                @click="productsOpen = !productsOpen"
              />
              <i
                :class="[
                  'header-button',
                  isProductsExpanded ? 'fas fa-angle-down' : 'fas fa-angle-up'
                ]"
                data-cy="productsTogglePanelExpanded"
                @click="productsExpanded = !productsExpanded"
              />
            </div>
          </div>
          <panel-products
            :table-max-height="computedTableMaxHeightProducts"
            :table-elements="computedTableMaxItemsProducts"
          ></panel-products>
        </div>
      </transition>
    </div>
    <div class="transition-filters">
      <transition name="slide" class="filters">
        <div
          :class="[
            'top-panel',
            isFiltersExpanded ? 'top-panel__expanded' : 'top-panel__reduced'
          ]"
        >
          <div
            data-cy="filtersPanel"
            :class="[
              'top-panel-body',
              isFiltersExpanded
                ? 'top-panel-body__expanded'
                : 'top-panel-body__reduced'
            ]"
          >
            <panel-filters :filters-width="computedSelectWidth"></panel-filters>
          </div>
          <div
            class="top-panel__footer"
            data-cy="filtersToggle"
            @click="filtersExpanded = !filtersExpanded"
          >
            <div class="top-panel__footer__icon">
              <object
                :data="require('@/assets/filters.svg')"
                type="image/svg+xml"
                style="pointer-events: none;"
              />
            </div>
            <label class="top-panel__footer__label">Filters Manager</label>
          </div>
        </div>
      </transition>
    </div>
  </div>
</template>
<script>
import { resizeHelperMixin } from "@/mixins/resize-helper.js";
import PanelProducts from "@/components/PanelProducts.vue";
import PanelSensors from "@/components/PanelSensors.vue";
import PanelFilters from "@/components/PanelFilters.vue";
import { mapGetters } from "vuex";
import * as types from "@/store/types";
export default {
  components: {
    "panel-products": PanelProducts,
    "panel-sensors": PanelSensors,
    "panel-filters": PanelFilters
  },
  mixins: [resizeHelperMixin],
  data() {
    return {
      filtersExpanded: false,
      productsOpen: false,
      productsExpanded: false,
      sensorsOpen: false,
      sensorsExpanded: false
    };
  },
  computed: {
    ...mapGetters({
      areProductsData: [types.G_GET_PRODUCTS_COUNT],
      areSensorsData: [types.G_GET_SENSORS_COUNT],
      areFiltersReady: [types.G_GET_FILTERS_READY]
    }),
    isFiltersExpanded() {
      return this.filtersExpanded;
    },
    isProductsOpen() {
      return this.productsOpen;
    },
    isProductsExpanded() {
      return this.productsExpanded;
    },
    isSensorsOpen() {
      return this.sensorsOpen;
    },
    isSensorsExpanded() {
      return this.sensorsExpanded;
    }
  },
  watch: {
    areFiltersReady(newValue) {
      this.filtersExpanded = newValue;
    },
    areProductsData(newValue) {
      this.productsOpen = newValue > 0 ? true : false;
    },
    areSensorsData(newValue) {
      this.sensorsOpen = newValue > 0 ? true : false;
    }
  },
  mounted() {
    this.setEventResize();
  }
};
</script>
<style scoped lang="scss">
.slide-enter-active,
.slide-leave-active {
  transition: transform 500ms ease;
}

.slide-enter,
.slide-leave-to {
  transition: all 500ms ease-in 0s;
}

.transition-sensors {
  .slide-enter,
  .slide-leave-to {
    transform: translateX(-100%);
  }
}

.transition-products {
  .slide-enter,
  .slide-leave-to {
    transform: translateX(+100%);
  }
}

.transition-filters {
  .slide-enter,
  .slide-leave-to {
    transform: translateY(-100%);
  }
}

.show-button {
  height: 6vh;
  position: fixed;
  z-index: 1; /* Needs to sit above the map menu icon */
  top: 60vh;
  border-radius: 50%;
  border: solid $color-primary;
  &:hover {
    cursor: pointer;
  }

  &__right {
    right: 0;
  }

  &__left {
    left: 0;
  }
}

.header-panel {
  padding: 1.5vw;

  .header-button {
    margin: 0 5px;
    &:hover {
      cursor: pointer;
    }
  }

  .header-buttons {
    margin: 0 5px;
    display: flex;
    &__expanded {
      margin-top: calc(1.5vh + 20px);
      margin-bottom: calc(-1.5vh - 20px);
    }
    &__right {
      flex-direction: row;
    }

    &__left {
      flex-direction: row-reverse;
    }
  }
}

.top-panel {
  border: 1.5px solid $color-light-grey;
  left: 20vw;
  width: 60vw;
  position: fixed;
  z-index: 2; /* Needs to sit above the menu icon */
  opacity: 1;
  background-image: $color-panels;
  border-radius: 0 0 45px 45px;
  display: flex;
  flex-direction: column;
  overflow: hidden;
  &__expanded {
    top: 0;
    transition: all 500ms ease-in 0s;
  }

  &__reduced {
    top: -15.5vh;
    transition: all 500ms ease-in 0s;
  }
  .top-panel-body {
    width: 100%;
    overflow: hidden;
    transition: all 500ms ease-in 0s;
    &__expanded {
      min-height: 0;
      max-height: 25vh;
      transition: all 500ms ease-in 0s;
    }

    &__reduced {
      min-height: 15vh;
      max-height: 15vh;
      transition: all 500ms ease-in 0s;
    }
  }

  &__footer {
    height: 25px;
    width: 100%;
    display: flex;
    flex-flow: row nowrap;
    justify-content: center;
    padding: 0.5rem;
    border-top: 1.5px solid $color-light-grey;

    &:hover {
      background: $color-light-grey;
      cursor: pointer;
    }
    &__icon {
      width: 32px;
      margin: 0 5px;
    }
    &__label {
      font-weight: bold;
      margin: 6px 5px;
      &:hover {
        cursor: pointer;
      }
    }
  }
}

.side-panel {
  border: 1.5px solid $color-light-grey;
  height: 40vh;
  top: 60vh;
  width: 40vw;
  position: fixed;
  z-index: 2; /* Needs to sit above the menu icon */
  opacity: 1;
  background-image: $color-panels;

  &__expanded {
    top: 0;
    height: 100vh;
    transition: all 500ms ease-in 0s;
  }

  &__reduced {
    transition: all 500ms ease-in 0s;
  }

  &__products {
    right: 0;
    border-radius: 45px 0 0 0;
    padding: 0 0.5vh 0 0.5vh;
  }

  &__sensors {
    left: 0;
    border-radius: 0 45px 0 0;
    padding: 0 0.5vh 0 0.5vh;
  }
}
</style>
