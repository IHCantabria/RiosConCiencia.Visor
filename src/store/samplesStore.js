import { defineStore } from "pinia";
import {
  getYears,
  getCampaignFilter,
  getDownloadFilters,
} from "@/utils/data-operator-utils.js";
import { CAMPAIGN_VALUES } from "@/config/data-config";
import { ADMIN_ROLE_ID } from "@/config/role-config.js";
import { shallowRef } from "vue";

export const useSamplesStore = defineStore("samplesStore", {
  state: () => {
    return {
      samples: null,
      samplesPicts: null,
      yearOptions: [],
      yearFilter: {},
      campaignOptions: [],
      campaignFilter: {},
      downloadOptions: [],
      downloadFilters: [],
      user: null,
      riverSectionHistoricData: {
        active: false,
        riverSectionId: null,
      },
      riversGeoJson: shallowRef(null),
    };
  },
  getters: {
    getSamplesFiltered() {
      return this.samples
        ? this.samples.filter(
            (sample) =>
              sample.year == this.yearFilter.name &&
              sample.idCampaign == this.campaignFilter.id,
          )
        : [];
    },
    getSamplesFilteredDownloadIds() {
      if (!this.samples) {
        return [];
      } else if (
        !this.downloadFilters.length ||
        (this.downloadFilters.length == 1 && this.downloadFilters[0].id == 0)
      ) {
        return this.samples.map((sample) => sample.id);
      } else {
        return this.samples
          .filter((sample) => this.getFiltersIds.includes(sample.idFilter))
          .map((sample) => sample.id);
      }
    },
    getSamplesPictsFilteredDownloadIds() {
      if (!this.samplesPicts) {
        return [];
      } else if (
        !this.downloadFilters.length ||
        (this.downloadFilters.length == 1 && this.downloadFilters[0].id == 0)
      ) {
        return this.samplesPicts.map((sample) => sample.id);
      } else {
        return this.samplesPicts
          .filter((sample) => this.getFiltersIds.includes(sample.idFilter))
          .map((sample) => sample.id);
      }
    },
    getFiltersIds() {
      return this.downloadFilters.length
        ? this.downloadFilters.map((filter) => filter.id)
        : [];
    },
    getIsFilterAll() {
      return this.downloadFilters[0].id == 0;
    },
    getUserToken() {
      return this.user ? this.user.token : "";
    },
    getUserLogged() {
      return this.user;
    },
    getAdminLogged() {
      return this.user && this.user.roleId == ADMIN_ROLE_ID;
    },
    getSamplesLoaded() {
      return this.samples;
    },
  },
  actions: {
    setSamples(operation) {
      if (!operation) return;
      this.samples = operation;
      const years = getYears(operation);
      const filterCampaign = getCampaignFilter(
        operation,
        years[0],
        CAMPAIGN_VALUES,
      );
      this.yearOptions = years;
      this.yearFilter = years[0];
      this.campaignOptions = CAMPAIGN_VALUES;
      this.campaignFilter = filterCampaign;
      const downloadOptions = getDownloadFilters(years, CAMPAIGN_VALUES);
      this.downloadOptions = downloadOptions;
      this.downloadFilters = [downloadOptions[0]];
    },
    setPictsSamples(operation) {
      this.samplesPicts = operation;
    },
    setRiversGeoJson(operation) {
      this.riversGeoJson = operation;
    },
    setUser(operation) {
      this.user = operation;
    },
    setSampleCampaignFilter(operation) {
      this.campaignFilter = operation;
    },
    setSampleYearFilter(operation) {
      this.yearFilter = operation;
    },
    setSampleDownloadFilters(operation) {
      this.downloadFilters = operation;
    },
    async fetchSamplesAdmin(operation) {
      this.setSamplesAdmin(operation);
    },
    async fetchSamples(operation) {
      this.setSamples(operation);
    },
    async fetchPictsSamples(operation) {
      this.setPictsSamples(operation);
    },
    async fetchUser(operation) {
      this.setUser(operation);
    },
    async setSampleCampaignFilterAction(operation) {
      this.setSampleCampaignFilter(operation);
    },
    async setSampleYearFilterAction(operation) {
      this.setSampleYearFilter(operation);
    },
    async setSampleDownloadFiltersAction(operation) {
      this.setSampleDownloadFilters(operation);
    },
    setRiverSectionHistoricData(idRiverSection) {
      if (!idRiverSection) {
        this.riverSectionHistoricData.active = false;
        this.riverSectionHistoricData.riverSectionId = null;
        return;
      }
      this.riverSectionHistoricData.active = true;
      this.riverSectionHistoricData.riverSectionId = idRiverSection;
    },
  },
});
