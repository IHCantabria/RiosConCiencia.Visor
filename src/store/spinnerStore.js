import { defineStore } from "pinia";
import { spinnerConfig, spinnerEnums } from "@/config/spinner-config";

export const useSpinnerStore = defineStore("spinnerStore", {
  state: () => {
    return {
      mlActive: false,
      calls: 0,
      spinnerConfig: spinnerConfig,
      spinnerEnums: spinnerEnums,
    };
  },
  getters: {
    mlTasksConfig() {
      return this.spinnerConfig?.tasksConfig || null;
    },
  },
  actions: {
    show() {
      this.calls++;
      this.mlActive = true;
    },
    hide() {
      this.calls--;
      if (this.calls == 0) {
        this.mlActive = false;
      }
      if (this.calls == 0) this._clearTasks();
    },
    configureModeSpinner() {
      this.spinnerConfig.mode = this.spinnerEnums.ModesSpinner.progress;
    },
    addTask(name, text) {
      this.spinnerConfig.tasksConfig.tasks.push({
        id: this.spinnerConfig.tasksConfig.tasks.length + 1,
        name,
        text,
        status: this.spinnerEnums.statusTask.pending,
      });
    },
    updateTask(name, status = null, text = null) {
      const indexTask = this.spinnerConfig.tasksConfig.tasks.findIndex(
        (task) => task.name === name,
      );
      if (status) {
        this.spinnerConfig.tasksConfig.tasks[indexTask].status = status;
      }
      if (text) {
        this.spinnerConfig.tasksConfig.tasks[indexTask].text = text;
      }
    },
    _clearTasks() {
      this.spinnerConfig.tasksConfig.tasks = [];
    },
  },
});
