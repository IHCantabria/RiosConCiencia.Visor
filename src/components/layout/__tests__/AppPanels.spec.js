import { shallowMount, createLocalVue } from "@vue/test-utils";
import AppPanels from "../AppPanels.vue";
import merge from "lodash.merge";
import Vuex from "vuex";
import * as types from "@/store/types";

const localVue = createLocalVue();
localVue.use(Vuex);

function createWrapper(overrides) {
  const defaultMountingOptions = {
    localVue
  };
  return shallowMount(AppPanels, merge(defaultMountingOptions, overrides));
}

describe("appPanels.vue", () => {
  let store;
  beforeEach(() => {
    store = new Vuex.Store({
      modules: {
        apiSensors: {
          getters: {
            [types.G_GET_SENSORS_COUNT]: () => true
          }
        },
        datahub: {
          getters: {
            [types.G_GET_PRODUCTS_COUNT]: () => true
          }
        },
        filters: {
          getters: {
            [types.G_GET_FILTERS_READY]: () => false
          }
        }
      }
    });
  });
  //UNIT TESTS
  it("init setEventResize called after mounted", () => {
    expect.assertions(1);
    const setEventResize = jest.fn();
    createWrapper({
      methods: {
        setEventResize
      },
      store
    });
    expect(setEventResize).toHaveBeenCalled();
  });
  it("computed isFiltersExpanded updated when filtersExpanded change", () => {
    expect.assertions(2);
    const wrapper = createWrapper({ store });
    expect(wrapper.vm.isFiltersExpanded).toBe(false);
    wrapper.setData({ filtersExpanded: true });
    expect(wrapper.vm.isFiltersExpanded).toBe(true);
  });
  it("computed isProductsOpen updated when productsOpen change", () => {
    expect.assertions(2);
    const wrapper = createWrapper({ store });
    expect(wrapper.vm.isProductsOpen).toBe(false);
    wrapper.setData({ productsOpen: true });
    expect(wrapper.vm.isProductsOpen).toBe(true);
  });
  it("computed isProductsExpanded updated when productsExpanded", () => {
    expect.assertions(2);
    const wrapper = createWrapper({ store });
    expect(wrapper.vm.isProductsExpanded).toBe(false);
    wrapper.setData({ productsExpanded: true });
    expect(wrapper.vm.isProductsExpanded).toBe(true);
  });
  it("computed isSensorsOpen updated when sensorsOpen", () => {
    expect.assertions(2);
    const wrapper = createWrapper({ store });
    expect(wrapper.vm.isSensorsOpen).toBe(false);
    wrapper.setData({ sensorsOpen: true });
    expect(wrapper.vm.isSensorsOpen).toBe(true);
  });
  it("computed isSensorsExpanded updated when sensorsExpanded", () => {
    expect.assertions(2);
    const wrapper = createWrapper({ store });
    expect(wrapper.vm.isSensorsExpanded).toBe(false);
    wrapper.setData({ sensorsExpanded: true });
    expect(wrapper.vm.isSensorsExpanded).toBe(true);
  });
  //SNAPSHOT TESTS
  it("renders correctly", () => {
    expect.assertions(1);
    const wrapper = createWrapper({ store });
    expect(wrapper.element).toMatchSnapshot();
  });
});
