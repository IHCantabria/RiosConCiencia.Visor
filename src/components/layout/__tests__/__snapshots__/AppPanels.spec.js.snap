// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`appPanels.vue renders correctly 1`] = `
<div
  class="app-info"
>
  <div
    class="panel-controls"
  >
    <div
      class="show-button show-button__left"
      data-cy="sensorsToggle"
      title="Sensors"
    >
      <object
        data=""
        style="height: 100%; pointer-events: none;"
        type="image/svg+xml"
      />
    </div>
     
    <div
      class="show-button show-button__right"
      data-cy="productsToggle"
      title="Products"
    >
      <object
        data=""
        style="height: 100%; pointer-events: none;"
        type="image/svg+xml"
      />
    </div>
  </div>
   
  <div
    class="transition-sensors"
  >
    <div
      class="sensors side-panel side-panel__sensors side-panel__reduced"
      data-cy="sensorsPanel"
      name="slide"
      style="display: none;"
    >
      <div
        class="header-panel"
      >
        <div
          class="header-buttons header-buttons__left"
        >
          <i
            class="fas fa-times header-button"
            data-cy="sensorsTogglePanel"
          />
           
          <i
            class="header-button fas fa-angle-up"
            data-cy="sensorsTogglePanelExpanded"
          />
        </div>
      </div>
       
      <panel-sensors-stub
        tableelements="15"
        tablemaxheight="171.83999999999997px"
      />
    </div>
  </div>
   
  <div
    class="transition-products"
  >
    <div
      class="products side-panel side-panel__products side-panel__reduced"
      data-cy="productsPanel"
      name="slide"
      style="display: none;"
    >
      <div
        class="header-panel"
      >
        <div
          class="header-buttons header-buttons__right"
        >
          <i
            class="fas fa-times header-button"
            data-cy="productsTogglePanel"
          />
           
          <i
            class="header-button fas fa-angle-up"
            data-cy="productsTogglePanelExpanded"
          />
        </div>
      </div>
       
      <panel-products-stub
        tableelements="15"
        tablemaxheight="171.83999999999997px"
      />
    </div>
  </div>
   
  <div
    class="transition-filters"
  >
    <div
      class="filters top-panel top-panel__reduced"
      name="slide"
    >
      <div
        class="top-panel-body top-panel-body__reduced"
        data-cy="filtersPanel"
      >
        <panel-filters-stub
          filterswidth="204.8px"
        />
      </div>
       
      <div
        class="top-panel__footer"
        data-cy="filtersToggle"
      >
        <div
          class="top-panel__footer__icon"
        >
          <object
            data=""
            style="pointer-events: none;"
            type="image/svg+xml"
          />
        </div>
         
        <label
          class="top-panel__footer__label"
        >
          Filters Manager
        </label>
      </div>
    </div>
  </div>
</div>
`;
